### ISSUE TITLE
[Bug] Every few months the version update hangs

### ISSUE THREAD

**Issue by coolaj86** [webinstall/webi-installers#534]
# Feb 13, 2023 Update

I've deployed #565 to production.

I'm very confident (after extensive testing) that this is the final fix to intermittent failure:
- fixes incorrect cache time from 5 **seconds** (debug value) to 5 **minutes** (production value)
- always returns release info as long as usable data is available (not reset on error)

**Jan 24, 2023 Update**
 - I've deployed #551 to production.
 - I believe this refactor of release info updates is the final solution for this problem.
 - Waiting another month to close the issue...

# Original

Over the past few years we've had 2 or 3 reports like this one:
See https://github.com/webinstall/webi-installers/issues/533

Something won't install because it just hangs, and the JSON version can't be fetched.

I think that we're either missing a timeout, or mishandling it.

Come to think of it, this may be due to https://github.com/therootcompany/request.js/pull/7 and may just need to update `@root/request`.

Thankfully the hangup is per-package and not everything at once, but I'd like to get to the bottom of it.

**Comment by wreszelewski** [comment]
Hello! It happens again for k9s. It looks exactly like this: https://github.com/webinstall/webi-installers/issues/533

**Comment by coolaj86** [comment]
Hmmm, well the previous "fix" (https://github.com/webinstall/webi-installers/pull/535) didn't work.

I've restarted the service. I'll investigate further over Christmas break.

**Comment by hwatts** [comment]
The k9s install is consistently hanging with no output at the moment

**Comment by ncstate01** [comment]
I'm seeing the same thing.

**Comment by coolaj86** [comment]
Restarted again just now.

I'll escalate this to high-priority. I will find a more permanent solution by this weekend at the latest.

**Comment by ncstate01** [comment]
It works again.  Thank you.

**Comment by coolaj86** [comment]
I've reworked the release update code.

I believe I discovered (and refactored away) a promise deadlock, and put in better failsafes for when a request hangs or fails.

It's now deployed via #551.

I'll leave this issue open for another month or two just to see if the issue arises again.

**Comment by MP91** [comment]
Seems to happen again. At least I'm not able to install k9s. https://webinstall.dev/api/releases/k9s.json opens but is empty...

Don't know what causes this.

**Comment by coolaj86** [comment]
Okay... I spent more time checking error conditions locally this time:
- https://github.com/webinstall/webi-installers/pull/565

Now as long as any usable data is available, the 500 will never propagate to the end user.

The worst possible outcome is that something stops getting release updates, but we don't know.

However, I'm almost certain that this was based on some sort of rate-limit trigger via GitHub Releases API.

I also increased the rate limit cache time, which was set far too low (I accidentally pushed the debug value of 5 **seconds** rather than 5 **minutes** last time).
